### ISSUE TITLE
[Enhancement] Icon search performance

### ISSUE THREAD

**Issue by effleurager** [Neodymium7/BetterDiscordStuff#68]
**Is your feature request related to a problem? Please describe.**
Loading the VoiceActivity plugin takes upwards of 15 seconds on my machine, spending the vast majority of time iterating over Webpack modules for each requested icon. This makes using Discord at startup an unpleasant wait.

**Describe the solution you'd like**
Replace individual search calls with a pattern matching any required icons, to reduce interation of all modules to a single-pass. The resulting subset can then be trivially searched.

**Describe alternatives you've considered**
- Bulk fetching modules describing an SVG image, then searching within the resultant list per requested icon
  - An improvement, but still slower and more memory-intensive than the proposed feature
- Dissecting the internals of Discord to find a direct mapping of icon to module
  - Requires an incredible time investment for something which may be a fragile solution

**Additional context**
Profile of plugin loading
![image](https://github.com/user-attachments/assets/8c18daa2-02b2-4fbd-a3d3-c231374ee807)

**Comment by effleurager** [comment]
Thanks for the phenomenally quick turnaround on this! 🤩

![Screenshot_20240822-134415](https://github.com/user-attachments/assets/8742e5aa-97d8-426b-9b98-f0c08233d016)

I've been trying to figure out the massive speed-up, but the [diff](https://github.com/Neodymium7/BetterDiscordStuff/compare/9ec96fde516b61c8a883bfc96fcdcdb34cc20216...eadbb4222c11c5fadd21bc70c1af1e13e7ea87ac) suggests it's something already solved within BetterDiscord?

**Comment by Neodymium7** [comment]
Yeah, no problem! I had been wondering myself about what was increasing the load times so much, and I hadn't actually thought about fetching the icons, so thanks for letting me know that was a potential issue!

In case you're curious what I changed:

Ideally, the optimal way to get the icons would be by searching for a string (like an SVG path) within the exported function's toString. A while ago when Discord started transitioning to their new icon set, all of the SVG icons were wrapped in a common function to determine which icon set to use, so the functions that included the SVG paths weren't exported themselves. The best alternative I thought of was to instead search the source code of the entire module, which is **_very_** slow, especially when you're doing it for every single module, and then for each icon.

Recently, a Discord update slowed down module searching a lot, which seemed to exponentially affect my already slow search methods (i.e. your 15 second loading times). Now that Discord has completely moved over to their new icon set and removed the wrapper function, that slower source search method was unnecessary. So, all I did to fix the loading times was move back to the previous, much faster search method. You can see that change [here](https://github.com/Neodymium7/BetterDiscordStuff/compare/9ec96fde516b61c8a883bfc96fcdcdb34cc20216...eadbb4222c11c5fadd21bc70c1af1e13e7ea87ac#diff-588156a749bd69eaedb5890a1c7b6071abc0b6f60a9e74216effcff4d64af06d). Kind of crazy how just a one line change can have such drastic results!

TL;DR I changed from a really, really slow module search method to a much better one for fetching the icons
